quick_sort.py

```diff
@@ -0,0 +1,63 @@
+def quisort(uslist, lo=None, hi=None):
+    """Sort in-place an unsorted list or slice of a list
+
+    lo and hi correspond to the start and stop indices for the list slice"""
+
+    if hi is None:
+        hi = len(uslist) - 1
+    if lo is None:
+        lo = 0
+
+    def partition(uslist, lo, hi):
+        """Compare and swap values over list slice"""
+        p = uslist[hi]
+        i = lo - 1
+        j = lo
+        while j < hi:
+            if uslist[j] <= p:
+                i = i + 1
+                uslist[i], uslist[j] = uslist[j], uslist[i]
+            j += 1
+        i += 1
+        uslist[i], uslist[hi] = uslist[hi], uslist[i]
+        return i
+
+    if lo < hi:
+        p = partition(uslist, lo, hi)
+        quisort(uslist, lo, p - 1)
+        quisort(uslist, p + 1, hi)
+
+
+if __name__ == "__main__":
+    """Test time performance for best and worst cases"""
+    import time
+    from random import shuffle
+
```

Comment:
Maybe inform the user that you are running a test here and it may take a moment. It just seems to freeze when you run the script. Obviously for this assignment we all know what is happening, but little details like this show attention to detail and UX (user experience).